tests/unit/test_runner_manager.py

```diff
@@ -271,7 +278,7 @@ def test_reconcile_issues_runner_installed_event(
     runner_manager: RunnerManager,
     monkeypatch: MonkeyPatch,
     issue_event_mock: MagicMock,
-    charm_state: MagicMock,
+    charm_state: CharmState,
```

Comment:
Mhh, this change does not seem to be correct. My IDE now also complains with an err msg
```
'CharmState' object attribute 'is_metrics_logging_available' is read-only 
```
What is the reasoning for this?

Reply:
This was error due to multiple refactor that removed then add back this arg.
It should be fixed now.